### PR TITLE
build(property-proxy): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution

### CODE REVIEW

File: experimental/PropertyDDS/packages/property-proxy/src/test/tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
-	"extends": "@fluidframework/build-common/ts-common-config.json",
+	"extends": "../../../../../../common/build/build-common/tsconfig.test.node16.json",
 	"compilerOptions": {
 		"rootDir": "./",
-		"outDir": "../../dist/test",
+		"outDir": "../../lib/test",
 		"types": ["node", "jest"],
-		"declaration": false,
-		"declarationMap": false,
 		"allowJs": true,
```
Comment (jason-ha) [inline experimental/PropertyDDS/packages/property-proxy/src/test/tsconfig.json:7]: If there are .js files in use, those are unlikely to be compatible in both CJS and ESM. Are there any? Are things working or not?

File: experimental/PropertyDDS/packages/property-proxy/jest.config.cjs [hunk 2]
```diff
@@ -19,10 +19,14 @@ module.exports = {
...
+		// Remove explicit .js from local paths to allow jest to find the .ts* files
+		"^(\\.{1,2}/.*)\\.js$": "$1",
+	},
 	globals: {
 		"ts-jest": {
 			tsconfig: "src/test/tsconfig.json",
 		},
 	},
-	testPathIgnorePatterns: ["/node_modules/", "dist"],
+	testPathIgnorePatterns: ["/node_modules/", "dist", "lib"],
```
Comment (jason-ha) [inline experimental/PropertyDDS/packages/property-proxy/jest.config.cjs:31]: Maybe. Is there a default match for jest tests? I don't see anything in this config.

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: experimental/PropertyDDS/packages/property-proxy/src/test/tsconfig.json (hunk 1 above)
Comment (RishhiB) [reply experimental/PropertyDDS/packages/property-proxy/src/test/tsconfig.json:7]: I tested setting this to false and deleting this line. both are able to build, build tests and pass tests.

Comment (jason-ha) [reply experimental/PropertyDDS/packages/property-proxy/src/test/tsconfig.json:7]: But are there any .js files? We can't really rely on tests. There might not be coverage.

Comment (RishhiB) [reply experimental/PropertyDDS/packages/property-proxy/src/test/tsconfig.json:7]: there are no js files

File: experimental/PropertyDDS/packages/property-proxy/jest.config.cjs (hunk 2 above)
Comment (tylerbutler) [reply experimental/PropertyDDS/packages/property-proxy/jest.config.cjs:31]: According to [the docs](https://jestjs.io/docs/configuration#testpathignorepatterns-arraystring), the default is `["/node_modules/"]`

Comment (tylerbutler) [reply experimental/PropertyDDS/packages/property-proxy/jest.config.cjs:31]: So assuming jest is supposed to test the .ts files, not the built JS files, then adding lib here is correct.

File: experimental/PropertyDDS/packages/property-proxy/src/test/tsconfig.json (hunk 1 above)
Comment (tylerbutler) [reply experimental/PropertyDDS/packages/property-proxy/src/test/tsconfig.json:7]: Based on that info, I think it's safe to remove that line from the config.

**Comment by sonalideshpandemsft** [comment]
clsoing in favor of https://github.com/microsoft/FluidFramework/pull/20313

**Pull request closed without merging**